### PR TITLE
libomptarget OpenCL plugin

### PR DESCRIPTION
**Pull request by ErnestoSu**
libomptarget OpenCL plugin

### CODE REVIEW

**Comment by bader** [comment]
@xtian-github, @vzakhari, @abhinavgaba, please, review and merge this pull request.

File: openmp/libomptarget/plugins/opencl/src/rtl.cpp [hunk 1]
```diff
@@ -0,0 +1,3625 @@
...
+#define NT_INTEL_ONEOMP_OFFLOAD_IMAGE_COUNT 2
+#define NT_INTEL_ONEOMP_OFFLOAD_IMAGE_AUX 3
+
+static bool isValidOneOmpImage(__tgt_device_image *Image,
+                               uint64_t &MajorVer,
+                               uint64_t &MinorVer) {
+  char *ImgBegin = reinterpret_cast<char *>(Image->ImageStart);
+  char *ImgEnd = reinterpret_cast<char *>(Image->ImageEnd);
+  size_t ImgSize = ImgEnd - ImgBegin;
+  ElfL E(ImgBegin, ImgSize);
```
Comment (vzakhari) [inline openmp/libomptarget/plugins/opencl/src/rtl.cpp:1423]: Does this change-set have to be buildable?  `ElfL` is defined in `plugins/common`.

Comment (ErnestoSu) [reply openmp/libomptarget/plugins/opencl/src/rtl.cpp:1423]: @vzakhari , the added files are not intended to be part of the build. The CMakeLists.txt is included for reference only. I'm relying on the fact that the parent-level openmp/libomptarget/plugins/CMakeLists.txt does not say "add_subdirectory(opencl)"; does that guarantee that the build will ignore stuff under plugins/opencl/ ?

**Comment by ErnestoSu** [comment]
I'm abandoning this pull request and replace it with an updated one.

**Pull request closed without merging**